### DISCUSSION TITLE
integrate DFTB+ with DeePMD-kit

### DISCUSSION THREAD

**Discussion by njzjz** [dftbplus/dftbplus#1178] [Ideas]
Dear developers,

I developed a machine learning potential model (powered by [DeePMD-kit](https://github.com/deepmodeling/deepmd-kit) to correct DFTB3 to the DFT level. This work is published today at https://doi.org/10.1021/acs.jctc.2c01172. I noticed that in #885, the ChIMES was added to DFTB+. I wonder if DFTB+ could accept a similiar PR to integrate with DeePMD-kit, and what's the best way to integrate them?
I am one of the developers for DeePMD-kit. DeePMD-kit has provided [the C API](https://docs.deepmodeling.com/projects/deepmd/en/master/api_c/api_c.html) in the latest development version.

Thanks.

**Comment by bhourahine** [comment]
Hello, a PR for this sounds very interesting. We should discuss exactly which quantities need to be exchanged and whether there are additional API features required. There is the beginnings of a general set of API bindings for attaching external models in development (the work so far has mostly focused on electronic hamiltonians, instead of total energy terms). Does your model require the whole structure to be provided, or local regions around atoms or bonds?

**Reply by njzjz** [reply]
Thanks. Only local regions around atoms are required. This is similar to what we did to integrate with LAMMPS.

**Reply by achyuttt** [reply]
Are there any updates on integrating DFTB+ with DeepPMD-kit?

**Reply by njzjz** [reply]
There are no updates, but AmberTools24 has integrated both software and supported QM/MM and non-periodic QM simulations.

**Reply by achyuttt** [reply]
I am currently researching on making an interatomic potential from DFTB+ calculations for multi-atomic systems. But when I looked at the output file of DFTB+ calculation, it prints the atomic position only at the end of calculation unlike in vc-md calculation of DFT.  Can you please suggest me if I can get data from DFTB+ in such a way that I can convert them to be implemented in DeePMD-kit for interatomic potential making ?
